source/tests/system/nixnet_lin_driver_api_tests.cpp

```diff
@@ -134,47 +154,46 @@ TEST_F(NiXnetLINDriverApiTests, LoopbackTestFromExample_FetchData_DataLooksReaso
   state_value.set_lin_schedule_change(0);
   EXPECT_SUCCESS(client::write_state(stub(), output_session, WRITE_STATE_LIN_SCHEDULE_CHANGE, state_value));
   EXPECT_SUCCESS(client::start(stub(), input_session, START_STOP_SCOPE_NORMAL));
-  output_value_vtr[0] = (f64)(5 % 10);
-  output_value_vtr[1] = (f64)((5 % 10) * 10);
-  auto write_signal_single_point_response = EXPECT_SUCCESS(client::write_signal_single_point(stub(), output_session, output_value_vtr));
+  output_value_vtr[0] = 5;
+  output_value_vtr[1] = 50;
 
+  auto write_signal_single_point_response = EXPECT_SUCCESS(client::write_signal_single_point(stub(), output_session, output_value_vtr));
+  std::this_thread::sleep_for(std::chrono::seconds(1));
   auto read_signal_single_point_response = EXPECT_SUCCESS(client::read_signal_single_point(stub(), input_session, NUM_SIGNALS_IN));
+
   EXPECT_SUCCESS(client::clear(stub(), output_session));
   EXPECT_SUCCESS(client::clear(stub(), input_session));
-
   EXPECT_EQ(output_value_vtr[0], read_signal_single_point_response.value_buffer()[0]) << "LIN1 and LIN2 must be connected together.";
   EXPECT_EQ(output_value_vtr[1], read_signal_single_point_response.value_buffer()[1]) << "LIN1 and LIN2 must be connected together.";
 }
 
-TEST_F(NiXnetLINDriverApiTests, WriteSignalXYFromExample_ReadSignalXYFromExample_SignalMatches)
+TEST_F(NiXnetLINDriverApiTestsWithHardware, LoopbackSignalXYTestFromExample_WriteAndReadSignalXY_SignalMatches)
```

Comment:
note: Signal and Sample data was incorrectly generated for `write` call. It is supposed to be 1D buffer containing `NUM_SAMPLE` values for each signal concatenated to each other.

Also `NUM_SIGNALS` was set to 2, but `CreateSession` was passing in only one signal name.